AudioEngine/PacketParser.cpp

```diff
@@ -146,6 +146,64 @@ void packetParser::parseData(string packet)
 				}
 			}
 		}
+		else if (strcmp(*it, "flange") == 0)
+		{
+			if (it + 1 != packetData.end())
+			{
+				if (strcmp(*(it + 2), "apply") == 0)
+				{
+					if (strcmp(*(it + 4), "true") == 0)
+					{
+						request.flange.apply = true;
+					}
+					else if (strcmp(*(it + 4), "false") == 0)
+					{
+						request.flange.apply = false;
+					}
+				}
+				if (strcmp(*(it + 6), "mix") == 0)
+				{
+					request.flange.mix = stof(*(it + 8));
+				}
+				if (strcmp(*(it + 10), "depth") == 0)
+				{
+					request.flange.depth = stof(*(it + 12));
+				}
+				if (strcmp(*(it + 14), "rate") == 0)
+				{
+					request.flange.rate = stof(*(it + 16));
+				}
+			}
+		}
+		else if (strcmp(*it, "pitchshift") == 0)
+		{
+			if (it + 1 != packetData.end())
+			{
+				if (strcmp(*(it + 2), "apply") == 0)
+				{
+					if (strcmp(*(it + 4), "true") == 0)
+					{
+						request.pitchshift.apply = true;
+					}
+					else if (strcmp(*(it + 4), "false") == 0)
+					{
+						request.pitchshift.apply = false;
+					}
+				}
+				if (strcmp(*(it + 6), "pitch") == 0)
+				{
+					request.pitchshift.pitch = stof(*(it + 8));
+				}
+				if (strcmp(*(it + 10), "fftsize") == 0)
+				{
+					request.pitchshift.fftsize = stof(*(it + 12));
+				}
+				if (strcmp(*(it + 14), "maxchannels") == 0)
+				{
+					request.pitchshift.maxchannels = stof(*(it + 16));
+				}
+			}
+		}

```

Comment:
An error occurs when attempting to parse through these effects. 
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stof
Aborted
Ensure that tests are written to catch these kinds of errors, or run them locally on your own device.

Reply:
Also the arguments given from the packet don't match what is expected from the parser. 

Example: The packet send from the web-app will send this upon selecting flange and pitch shift.
'flange': {'apply': 'on', 'mix': '0.5', 'depth': '0.5', 'rate': '0.5'}, 'pitchshift': {'apply': 'on', 'pitch': '0.5', 'fftsize': 1024, 'maxchannels': 0, 'fft': '0.5', 'channels': '0.5'}}}

When on line 182 of the packetParser.cpp it appears that the parser is looking for values of "true" or "false" from the packet, but the parameters sent from the web-app are "on". Meaning when requested these effects will never get applied.


Lastly, the web app has a bug that occurs when you select the flange and pitch shift effects. If you select them the first time they go from "false" to "on" when you send a packet. However, If you remove the effects from selection and send the packet again they are still set to "on" in the data packet. Here are some screenshots of this occuring.
![effects_bug_1](https://user-images.githubusercontent.com/43712723/83346592-a7c8ad80-a2d2-11ea-85b0-4e238b011529.png)
![effects_bug_2](https://user-images.githubusercontent.com/43712723/83346593-ab5c3480-a2d2-11ea-9107-a99a2aafd10a.png)

Reply:
Can you verify if the "false to on" issue only happens on the flange/pitch effects, and not on the echo/eq effects? I based the code for the new effects on what was already written before.

Reply:
> Can you verify if the "false to on" issue only happens on the flange/pitch effects, and not on the echo/eq effects? I based the code for the new effects on what was already written before.

I just verified that this issue only effects the newer effects. Putting this issue up.